### PR TITLE
Fix/augmentations swap entities

### PR DESCRIPTION
**Pull request by ArshaanNazir**
## Checklist:

- [x] I've added Google style docstrings to my code.
- [x] I've used `pydantic` for typing when/where necessary.
- [x] I have linted my code
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by ArshaanNazir** [comment]
This PR deals with solving issues which we faced in augmenting swap entities for clinical files

File: langtest/transform/robustness.py [hunk 1]
```diff
@@ -491,6 +491,8 @@ def transform(
 
                 ent_start_pos = [1 if label[0] == "B" else 0 for label in sample_labels]
                 ent_idx = [i for i, value in enumerate(ent_start_pos) if value == 1]
+                if not ent_idx:
+                    continue
```
Comment (JulesBelveze) [inline langtest/transform/robustness.py:496]: I'm not sure how is this possible?
We do a check a few lines above that not all tokens are "O" -> so there should be at least one "B" right?

File: langtest/transform/robustness.py [hunk 2]
```diff
@@ -503,14 +505,27 @@ def transform(
...
                 replace_token = sent_tokens[replace_idx : replace_idx + len(replace_idxs)]
+                token_length = len(replace_token)
                 replace_token = " ".join(replace_token)
-
-                chosen_ent = random.choice(terminology[ent_type])
+                filtered_ents = [
+                    ent
+                    for ent in terminology[ent_type]
+                    if len(ent.split(" ")) == token_length
+                ]
```
Comment (JulesBelveze) [inline langtest/transform/robustness.py:514]: This means that we only swap entities with the same number of words right?

File: langtest/transform/robustness.py [hunk 3]
```diff
@@ -503,14 +505,27 @@ def transform(
...
+                if not filtered_ents:
+                    chosen_ent = replace_token
+                else:
+                    chosen_ent = random.choice(filtered_ents)
```
Comment (JulesBelveze) [inline langtest/transform/robustness.py:518]: Not sure to get what's cooking here..

**Review by JulesBelveze — COMMENTED** [review]
I'm thinking more and more of storing the list of tokens into `NERSample`.. since we have a lit of per-token labels this would be helpful and less error prone

Btw, @ArshaanNazir could you had a test with the `Sample` or edge case that was causing the augmentation to fail? (if you easily have access to it)

File: langtest/transform/robustness.py (hunk 3 above)
Comment (ArshaanNazir) [reply langtest/transform/robustness.py:518]: We look if we have same length replacement (that are normally there) If not there, we choose randomly. However while doing it randomly, we found that we were getting absurd replacements.

File: langtest/transform/robustness.py (hunk 1 above)
Comment (ArshaanNazir) [reply langtest/transform/robustness.py:496]: Yes, you are right. We can remove it

File: langtest/transform/robustness.py (hunk 2 above)
Comment (JulesBelveze) [reply langtest/transform/robustness.py:512]: @ArshaanNazir couldn't we make it more general?

**Review by JulesBelveze — COMMENTED** [review]
LGTM, could you please add a couple of tests @ArshaanNazir ?

**Review by chakravarthik27 — APPROVED** [review]
